// tori/lib/eventBus.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventBus = void 0;
/**
 * Simple wrapper of event bus.
 */
var EventBus = /** @class */ (function () {
    function EventBus() {
        this.listeners = {};
    }
    EventBus.prototype.on = function (event, listener) {
        var _this = this;
        if (!this.listeners[event]) {
            this.listeners[event] = [];
        }
        this.listeners[event].push(listener);
        return {
            remove: function () {
                _this.off(event, listener);
            },
        };
    };
    EventBus.prototype.once = function (event, listener) {
        var _this = this;
        this.listeners[event] = [listener];
        return {
            remove: function () {
                _this.off(event, listener);
            },
        };
    };
    EventBus.prototype.off = function (event, listener) {
        if (!listener) {
            delete this.listeners[event];
        }
        else if (this.listeners[event]) {
            var index = this.listeners[event].findIndex(function (item) { return item === listener; });
            if (~index) {
                this.listeners[event].splice(index, 1);
            }
        }
    };
    EventBus.prototype.emit = function (event) {
        var args = [];
        for (var _i = 1; _i < arguments.length; _i++) {
            args[_i - 1] = arguments[_i];
        }
        if (this.listeners[event]) {
            this.listeners[event].forEach(function (listener) {
                listener.apply(void 0, args);
            });
        }
    };
    return EventBus;
}());
exports.EventBus = EventBus;
exports.default = new EventBus();
